torch/fx/passes/constructor_mover_pass.py

```diff
@@ -0,0 +1,275 @@
+from collections import defaultdict
+from typing import Dict, Iterable, List, Optional, Set, Tuple
+
+import torch
+import torch._ops
+import torch._subclasses.fake_tensor
+import torch.fx
+import torch.utils._pytree as pytree
+
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+
+
+__all__ = [
+    "ZeroOrMultipleDevicesError",
+    "ConstructorMoverPass",
+]
+
+
+@torch.fx._compatibility.compatibility(is_backward_compatible=False)
+class ZeroOrMultipleDevicesError(RuntimeError):
```

Comment:
I see that there are chances to the code and refactorings. Can you please submit this as a ghstack with just the refactorings as the first pr and code changes as the second ?